dns_test.go

```diff
@@ -374,17 +374,47 @@ func TestDNSProtocol(t *testing.T) {
 
 		_, port, _ := net.SplitHostPort(addr.String())
 
-		// Force IPv4
+		// Default DNS Server port
 		module := Module{
 			Timeout: time.Second,
 			DNS: DNSProbe{
 				QueryName: "example.com",
-				Protocol:  protocol + "4",
+				Protocol:  protocol,
 			},
 		}
 		recorder := httptest.NewRecorder()
 		registry := prometheus.NewRegistry()
-		result := probeDNS(net.JoinHostPort("localhost", port), recorder, module, registry)
+		result := probeDNS("localhost", recorder, module, registry)
+		if result {
+			t.Fatalf("DNS protocol: \"%v\" with localhost:53 succeeded, expected failure.", protocol)
```

Comment:
Aha.. I introduced a new way to set default port via `DefaultTargetPort` argument in config.